cms/djangoapps/contentstore/views/item.py

```diff
@@ -631,10 +665,77 @@ def _create_item(request):
     )
 
     return JsonResponse(
-        {"locator": unicode(created_block.location), "courseKey": unicode(created_block.location.course_key)}
+        {'locator': unicode(created_block.location), 'courseKey': unicode(created_block.location.course_key)}
     )
 
 
+def _get_source_index(source_usage_key):
+    """
+    Get source index position of the XBlock.
+
+    Arguments:
+        source_usage_key (BlockUsageLocator): Locator of source item.
+
+    Returns:
+        source_index (int): Index position of the xblock in a parent.
+    """
+    store = modulestore()
+    source_item = store.get_item(source_usage_key)
+    source_parent = source_item.get_parent()
+    return source_parent.children.index(source_usage_key)
+
+
+def is_valid_move(source_usage_key, dest_parent_usage_key):
```

Comment:
I think a better option would be to change ``_move_item`` to do the assertions itself, because then the checks are right by the code that is going to perform the change.